sherpa/astro/ui/tests/test_astro_ui_utils_simulation.py

```diff
@@ -107,6 +107,75 @@ def test_fake_pha_incompatible_rmf(idval, clean_astro_ui):
         ui.fake_pha(idval, arf, rmf, 1000.0)
 
 
+@pytest.mark.parametrize("idval", [None, 1, "faked"])
+def test_fake_pha_exposure_is_none_data_is_none(idval, clean_astro_ui, reset_seed):
+    """What happens if the exposure is None in the fake_pha call and the DataPHA?"""
+
+    np.random.seed(7239652)
+
+    channels = np.arange(1, 4, dtype=np.int16)
+    counts = np.ones(3, dtype=np.int16)
+
+    # 0.5 keV bin widths mean that source evaluation is half the
+    # c0 value for a const1d model.
+    #
+    egrid = np.arange(2, 6) * 0.5
+    elo = egrid[:-1]
+    ehi = egrid[1:]
+    arf = ui.create_arf(elo, ehi)
+    rmf = ui.create_rmf(elo, ehi, e_min=elo, e_max=ehi)
+
+    ui.load_arrays(idval, channels, counts, ui.DataPHA)
+    ui.set_rmf(idval, rmf)
+    ui.set_arf(idval, arf)
+
+    cpt = ui.create_model_component('const1d', 'cpt')
+    cpt.c0 = 24
+    ui.set_source(idval, cpt)
+
+    ui.fake_pha(idval, arf=None, rmf=None, exposure=None)
```

Comment:
What is it even supposed to mean if `exposure=None`? It makes no sense to run a simulation for an undefined exposure. I *might* be convinced that we could default to the exposure time of previously loaded data, but if both are `None` what could that possibly do? I don't see any use case for that and I'm in fact surprised that that does not error out with something like "can't multiply array with None` or similar. 

At the very least, I would think that the behavior is probably undefined (not because the code makes it undefined, but because the question asked "fake a spectrum, but I won't tell you how long to fake it" is ill defined. It's almost analogous to "fake a spectrum, but I won't tell you what the size of the telescope is (no ARF) or what the input model is. Neither of those makes sense and gives a result, but apparently this does. Unless I'm missing something fundamental, I think it's a bug and any non-numerical value for the exposure time shoudl be disallowed.

[Note: Because @DougBurke put a lot of effort into sorting his commits, I'm trying now to review commit by commit. It's entirely possible that this changes again in the next commit, which is why I usually review the final PR and not individual commits. I'll find out which strategy causes more confusion...]

Reply:
So, we have a dataset identifier which can be used to define things like the channels, the response, the model, the exposure time. So I claim that `exposure=None` means "take the exposure time from the existing data", just like `rmf=None` does.

It depends if you are saying "I want to simulate the existing data (e.g. to see how reliable the results are)" or "I want to see what happens with a longer exposure". I think both are valid and should really be supported.

Reply:
OK. I'll look at the docs to see if I should have known that.

Reply:
Also note that there is meant to be more-exstensive doc changes in the follow-on PR #1684, which might be a better place to improve things.

Reply:
see https://github.com/DougBurke/sherpa/pull/4 - but if that's part of #1684 (which I have not looked at yet), that's fine, too.